lib/article_json/export/apple_news/elements/embed.rb

```diff
@@ -0,0 +1,96 @@
+module ArticleJSON
+  module Export
+    module AppleNews
+      module Elements
+        class Embed < Base
+          # Embed
+          # @return [Hash]
+          def export
+            {
+              role: role,
+              URL: source_url,
+              caption: caption
+            }.compact
+          end
+
+          private
+
+          def role
+            @role ||=
+              case embed_type
+              when :youtube_video, :vimeo_video, :dailymotion_video
+                :embedwebvideo
+              when :facebook_video
+                :facebook_post
+              when :tweet
+                :tweet
+              when :slideshare
+                nil
+              when :soundcloud
+                nil
+              else
+                nil
+              end
+          end
+
+          def source_url
+            case embed_type
+            when :youtube_video
+              build_embeded_youtube_url
+            when :vimeo_video
+              build_embeded_vimeo_url
+            when :dailymotion_video
+              build_embeded_vimeo_url
+            when :facebook_video
+              build_facebook_video_url
+            when :tweet
+              build_twitter_url
+            when :slideshare
+              nil
+            when :soundcloud
+              nil
+            else
+              nil
+            end
+          end
+
+          def caption
+            return nil if role.nil?
+
+            @element.caption.first&.content
+          end
+
+          def build_embeded_youtube_url
+            "https://www.youtube.com/embed/#{embed_id}"
+          end
+
+          def build_embeded_vimeo_url
+            "https://player.vimeo.com/video/#{embed_id}"
+          end
+
+          def build_embeded_dailymotion_url
+            "https://geo.dailymotion.com/player.html?video=#{embed_id}"
+          end
+
+          def build_facebook_video_url
+            username, id = embed_id.to_s.split("/")
```

Comment:
you can also tell into how many items you want to split the string: `split("/", 2)`

Reply:
Ah, good to know. I can add that.

Reply:
Nice, I didn't know about the second parameter!